File: CH5/Scope/execdef.js
function foo(n) {
    if (n%2==0) {
      function bar(n) {
        return 2*n;
      }
    }
    else {
      function bar(n) {
        return 2*n+1;
      }
    }
    return bar(n);
  }
  number = prompt("Enter a number: ", 0);
  foo_number = foo(number);
  alert ("number is "+number+" foo(number) is "+foo_number);
  // returns 2n if even, 2n+1 otherwise

File: CH5/Scope/execdef.py
def foo(n):
  if (n%2 ==0):
    def bar(n):
      return 2*n
  else:
    def bar(n):
      return 2*n+1
  return bar(n)

number= int(input("Enter a number: "))
foo_number = foo(number)
print("number is "+str(number)+" foo(number) is "+str(foo_number))

"""
print("bar(n)"+str(bar(number)))
"""
